engine/flink/components-api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkIntermediateRawSource.scala

```diff
@@ -20,54 +20,94 @@ import pl.touk.nussknacker.engine.flink.api.compat.ExplicitUidInOperatorsSupport
-  *  2. sets UID
+  *  2. sets UID and overrides source name
   *  3. assigns timestamp and watermarks
   *  4. initializes Context that is streamed within output DataStream (here raw data are mapped to Context)
   * It separates raw event data produced by SourceFunction and data released to the stream as Context variables.
   * By default it uses basic "single input value" implementation of initializer, see [[BasicContextInitializer]].
   *
   * @tparam Raw - type of raw event that is generated by flink source function.
   */
-trait FlinkIntermediateRawSource[Raw] extends ExplicitUidInOperatorsSupport { self: Source =>
+trait FlinkIntermediateRawSource[Raw]
+    extends CustomizableContextInitializerSource[Raw]
+    with CustomizableTimestampWatermarkHandlerSource[Raw]
+    with ExplicitTypeInformationSource[Raw] { self: Source =>
 
-  // We abstracting to stream so theoretically it shouldn't be defined on this level but:
-  // * for test mechanism purpose we need to know what type will be generated.
-  // * for production sources (eg BaseFlinkSource, KafkaSource) it is used to determine type information for flinkSourceFunction
-  def typeInformation: TypeInformation[Raw]
-
-  def timestampAssigner: Option[TimestampWatermarkHandler[Raw]]
-
-  val contextInitializer: ContextInitializer[Raw] = new BasicContextInitializer[Raw](Unknown)
+  override def contextInitializer: ContextInitializer[Raw] = new BasicContextInitializer[Raw](Unknown)
 
   @silent("deprecated")
   def prepareSourceStream(
       env: StreamExecutionEnvironment,
       flinkNodeContext: FlinkCustomNodeContext,
       sourceFunction: SourceFunction[Raw]
   ): DataStream[Context] = {
+    FlinkIntermediateRawSourceUtils.prepareSourceStream(
+      env,
+      flinkNodeContext,
+      sourceFunction,
+      typeInformation,
+      timestampAssigner,
+      contextInitializer
+    )
+  }
+
+}
+
+object FlinkIntermediateRawSourceUtils {
+
+  @silent("deprecated")
+  def prepareSourceStream[Raw](
+      env: StreamExecutionEnvironment,
+      flinkNodeContext: FlinkCustomNodeContext,
+      sourceFunction: SourceFunction[Raw],
+      typeInformation: TypeInformation[Raw],
+      timestampWatermarkHandler: Option[TimestampWatermarkHandler[Raw]],
+      contextInitializer: ContextInitializer[Raw]
+  ): DataStream[Context] = {
+    val source = FlinkStandardSourceUtils.createSource(env, sourceFunction, typeInformation)
+    FlinkStandardSourceUtils.prepareSource(source, flinkNodeContext, timestampWatermarkHandler, contextInitializer)
+  }
+
+}
+
+object FlinkStandardSourceUtils extends ExplicitUidInOperatorsSupport {
+
+  @silent("deprecated")
+  def createSource[Raw](
+      env: StreamExecutionEnvironment,
+      sourceFunction: SourceFunction[Raw],
+      typeInformation: TypeInformation[Raw]
+  ): DataStreamSource[Raw] = {
+    env.addSource[Raw](sourceFunction, typeInformation)
+  }
+
+  def prepareSource[Raw](
+      source: DataStreamSource[Raw],
```

Comment:
Let's name it `sourceStream` or `flinkSource`. We have nussknacker sources (components) and I know ppl who would punish us for mixing those terms :)

Reply:
I've done some naming refactors:
1. `FlinkSource` - renamed `sourceStream` into `contextStream` that returns `DataStream[Context]`
2. `StandardFlinkSource` - has just a `sourceStream` that returns `DataStreamSource[Raw]` 
3. `StandardFlinkSource` implements `contextStream` with final so no one can think to override it - if he wants to override that means he should use `FlinkSource` directly and the api will reflect that

I think now its pretty clear what `StandardFlinkSource` does. I've also noted these changes in Migration Guide and I've updated the `FlinkComponents.md` docs.  I've decided to make some changes in that doc since I put some effort into understanding the sources and wanted to get more value out of it. Or if you think the changes will require much discussion I can revert into just a simple update.

Reply:
It looks pretty nice now. God job